tests/rptest/tests/acls_test.py

```diff
@@ -86,7 +86,7 @@ def prepare_cluster(self,
                 self.security.principal_mapping_rules = principal_mapping_rules
             self.redpanda.add_extra_rp_conf({
                 'kafka_mtls_principal_mapping_rules':
-                self.security.principal_mapping_rules
+                [self.security.principal_mapping_rules]
```

Comment:
from what i can tell it looks like we have instances of strings that contain multiple rules and this takes one of those multi-rule-strings and generates a rules list of size one. unless i'm missing something we don't have a test where we have a configuration of containing a list of more than one rule string?

Reply:
See `BOOST_AUTO_TEST_CASE(test_mtls_parsing_with_multiline)` in this commit

Reply:
that looks like it tests the parsing, but not an e2e test that ducktape allows us to do?